packages/central-server/src/apiV2/authenticate/authenticate.js

```diff
@@ -140,11 +140,15 @@ export async function authenticate(req, res) {
       permissionGroups: permissionGroupsByCountryId,
     });
 
-    // Reset rate limiting on successful authorisation
-    await consecutiveFailsRateLimiter.resetFailedAttempts(req);
-    await bruteForceRateLimiter.resetFailedAttempts(req);
-
     respond(res, authorizationObject, 200);
+
+    // Reset rate limiting on successful authorisation
+    setTimeout(async () => {
+      await Promise.all([
+        consecutiveFailsRateLimiter.resetFailedAttempts(req),
+        bruteForceRateLimiter.resetFailedAttempts(req),
+      ]);
+    }, 0);
```

Comment:
```suggestion
    // Reset rate limiting on successful authorisation, no need to block response by awaiting though 
    consecutiveFailsRateLimiter.resetFailedAttempts(req);
    bruteForceRateLimiter.resetFailedAttempts(req);
```

That looks totally fine to me @jaskfla :+1: In this scenario, I tend to go with the approach of just not awaiting the promise rather than calling `setTimeout`. The performance isn't quite matching I believe, and set timeout more explicitly ensures that the task will be scheduled separately, but I find the readability of just not awaiting to be a bit nicer. Happy either way!

On the downside of the error not being caught, that is true and it can make troubleshooting pretty annoying. I believe an error will still show up in the logs (`Unhandled promise rejection...`) but the real weirdness for the user would be if they try and login again after their successful login and get rejected with `Max login attempts: You must wait x seconds before reattempting`. That would only occur if they're attempting to login again within the `retryAfter` time windows, so it would be a fairly corner case...